fvm/evm/stdlib/contract.go

```diff
@@ -38,6 +42,491 @@ var evmAddressBytesType = sema.NewConstantSizedType(nil, sema.UInt8Type, types.A
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.PrimitiveStaticTypeInt256:
+					typ, err := gethABI.NewType("int256", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				}
+
+				return true
+			})
+
+			decoded, err := arguments.Unpack(data)
+			if err != nil {
+				panic(err)
+			}
+
+			values := make([]interpreter.Value, 0)
+			for _, arg := range decoded {
+				switch value := arg.(type) {
+				case string:
+					values = append(values, interpreter.NewStringValue(
+						inter,
+						common.NewStringMemoryUsage(len(value)),
+						func() string {
+							return value
+						},
+					))
+				case uint8:
+					values = append(values, interpreter.NewUInt8Value(
+						inter,
+						func() uint8 {
+							return value
+						},
+					))
+				case uint16:
+					values = append(values, interpreter.NewUInt16Value(
+						inter,
+						func() uint16 {
+							return value
+						},
+					))
+				case uint32:
+					values = append(values, interpreter.NewUInt32Value(
+						inter,
+						func() uint32 {
+							return value
+						},
+					))
+				case uint64:
+					values = append(values, interpreter.NewUInt64Value(
+						inter,
+						func() uint64 {
+							return value
+						},
+					))
+				case *big.Int:
+					values = append(values, interpreter.NewUInt128ValueFromBigInt(
+						inter,
+						func() *big.Int {
+							return value
+						},
+					))
+				case int8:
+					values = append(values, interpreter.NewInt8Value(
+						inter,
+						func() int8 {
+							return value
+						},
+					))
+				case int16:
+					values = append(values, interpreter.NewInt16Value(
+						inter,
+						func() int16 {
+							return value
+						},
+					))
+				case int32:
+					values = append(values, interpreter.NewInt32Value(
+						inter,
+						func() int32 {
+							return value
+						},
+					))
+				case int64:
+					values = append(values, interpreter.NewInt64Value(
+						inter,
+						func() int64 {
+							return value
+						},
+					))
+				case bool:
+					values = append(values, interpreter.BoolValue(value))
+				case gethCommon.Address:
+					var address types.Address
+					copy(address[:], value.Bytes())
+					values = append(values, EVMAddressToAddressBytesArrayValue(inter, address))
+				case []uint64:
+					arrayType := interpreter.NewVariableSizedStaticType(
+						inter,
+						interpreter.NewPrimitiveStaticType(
+							inter,
+							interpreter.PrimitiveStaticTypeUInt64,
+						),
+					)
+					arrValues := make([]interpreter.Value, 0)
+					for _, v := range value {
+						arrValues = append(arrValues, interpreter.NewUInt64Value(inter, func() uint64 {
+							return v
+						}))
+					}
+
+					arr := interpreter.NewArrayValue(
+						inter,
+						invocation.LocationRange,
+						arrayType,
+						common.ZeroAddress,
+						arrValues...,
+					)
+					values = append(values, arr)
+				}
```

Comment:
Add a default case and report an error. I think that should never occur, so might have to be an implementation error, because the Cadence type to geth type conversion before the unpack should already  reject invalid types? 🤔 (not 100% sure)